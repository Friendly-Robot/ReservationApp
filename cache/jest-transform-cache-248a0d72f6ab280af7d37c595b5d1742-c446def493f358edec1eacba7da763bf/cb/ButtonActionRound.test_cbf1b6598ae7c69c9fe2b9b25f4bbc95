f5cef999500a53a640ee5201e331aeed
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

require("react-native");

var _react = _interopRequireDefault(require("react"));

var _ButtonActionRound = _interopRequireDefault(require("../ButtonActionRound"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

describe('<ButtonActionRound />', function () {
  it('should be defined', function () {
    expect(_ButtonActionRound.default).toBeDefined();
  });
  it('renders correctly', function () {
    var tree = _reactTestRenderer.default.create(_react.default.createElement(_ButtonActionRound.default, null)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkJ1dHRvbkFjdGlvblJvdW5kLnRlc3QuanMiXSwibmFtZXMiOlsiZGVzY3JpYmUiLCJpdCIsImV4cGVjdCIsIkJ1dHRvbkFjdGlvblJvdW5kIiwidG9CZURlZmluZWQiLCJ0cmVlIiwicmVuZGVyZXIiLCJjcmVhdGUiLCJ0b0pTT04iLCJ0b01hdGNoU25hcHNob3QiXSwibWFwcGluZ3MiOiI7O0FBSUE7O0FBQ0E7O0FBQ0E7O0FBR0E7O0FBRUFBLFFBQVEsQ0FBQyx1QkFBRCxFQUEwQixZQUFNO0FBQ3RDQyxFQUFBQSxFQUFFLENBQUMsbUJBQUQsRUFBc0IsWUFBTTtBQUM1QkMsSUFBQUEsTUFBTSxDQUFDQywwQkFBRCxDQUFOLENBQTBCQyxXQUExQjtBQUNELEdBRkMsQ0FBRjtBQUlBSCxFQUFBQSxFQUFFLENBQUMsbUJBQUQsRUFBc0IsWUFBTTtBQUM1QixRQUFNSSxJQUFJLEdBQUdDLDJCQUFTQyxNQUFULENBQWdCLDZCQUFDLDBCQUFELE9BQWhCLEVBQXVDQyxNQUF2QyxFQUFiOztBQUNBTixJQUFBQSxNQUFNLENBQUNHLElBQUQsQ0FBTixDQUFhSSxlQUFiO0FBQ0QsR0FIQyxDQUFGO0FBSUQsQ0FUTyxDQUFSIiwic291cmNlc0NvbnRlbnQiOlsiLyoqXG4gKiBAZm9ybWF0XG4gKi9cblxuaW1wb3J0ICdyZWFjdC1uYXRpdmUnO1xuaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCBCdXR0b25BY3Rpb25Sb3VuZCBmcm9tICcuLi9CdXR0b25BY3Rpb25Sb3VuZCc7XG5cbi8vIE5vdGU6IHRlc3QgcmVuZGVyZXIgbXVzdCBiZSByZXF1aXJlZCBhZnRlciByZWFjdC1uYXRpdmUuXG5pbXBvcnQgcmVuZGVyZXIgZnJvbSAncmVhY3QtdGVzdC1yZW5kZXJlcic7XG5cbmRlc2NyaWJlKCc8QnV0dG9uQWN0aW9uUm91bmQgLz4nLCAoKSA9PiB7XG4gIGl0KCdzaG91bGQgYmUgZGVmaW5lZCcsICgpID0+IHtcbiAgICBleHBlY3QoQnV0dG9uQWN0aW9uUm91bmQpLnRvQmVEZWZpbmVkKCk7XG4gIH0pO1xuXG4gIGl0KCdyZW5kZXJzIGNvcnJlY3RseScsICgpID0+IHtcbiAgICBjb25zdCB0cmVlID0gcmVuZGVyZXIuY3JlYXRlKDxCdXR0b25BY3Rpb25Sb3VuZCAvPikudG9KU09OKCk7XG4gICAgZXhwZWN0KHRyZWUpLnRvTWF0Y2hTbmFwc2hvdCgpO1xuICB9KTtcbn0pXG4iXX0=