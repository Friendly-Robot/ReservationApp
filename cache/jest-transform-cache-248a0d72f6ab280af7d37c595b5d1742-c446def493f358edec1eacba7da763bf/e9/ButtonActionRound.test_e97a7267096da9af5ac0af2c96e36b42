05f705948ee078b7ce0a1f29395f6a16
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

require("react-native");

var _react = _interopRequireDefault(require("react"));

var _ButtonActionRound = _interopRequireDefault(require("../ButtonActionRound"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

it('renders correctly', function () {
  var tree = _reactTestRenderer.default.create(_react.default.createElement(_ButtonActionRound.default, null)).toJSON();

  expect(tree).toMatchSnapshot();
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkJ1dHRvbkFjdGlvblJvdW5kLnRlc3QuanMiXSwibmFtZXMiOlsiaXQiLCJ0cmVlIiwicmVuZGVyZXIiLCJjcmVhdGUiLCJ0b0pTT04iLCJleHBlY3QiLCJ0b01hdGNoU25hcHNob3QiXSwibWFwcGluZ3MiOiI7O0FBSUE7O0FBQ0E7O0FBQ0E7O0FBR0E7O0FBRUFBLEVBQUUsQ0FBQyxtQkFBRCxFQUFzQixZQUFNO0FBQzVCLE1BQU1DLElBQUksR0FBR0MsMkJBQVNDLE1BQVQsQ0FBZ0IsNkJBQUMsMEJBQUQsT0FBaEIsRUFBdUNDLE1BQXZDLEVBQWI7O0FBQ0FDLEVBQUFBLE1BQU0sQ0FBQ0osSUFBRCxDQUFOLENBQWFLLGVBQWI7QUFDRCxDQUhDLENBQUYiLCJzb3VyY2VzQ29udGVudCI6WyIvKipcbiAqIEBmb3JtYXRcbiAqL1xuXG5pbXBvcnQgJ3JlYWN0LW5hdGl2ZSc7XG5pbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IEJ1dHRvbkFjdGlvblJvdW5kIGZyb20gJy4uL0J1dHRvbkFjdGlvblJvdW5kJztcblxuLy8gTm90ZTogdGVzdCByZW5kZXJlciBtdXN0IGJlIHJlcXVpcmVkIGFmdGVyIHJlYWN0LW5hdGl2ZS5cbmltcG9ydCByZW5kZXJlciBmcm9tICdyZWFjdC10ZXN0LXJlbmRlcmVyJztcblxuaXQoJ3JlbmRlcnMgY29ycmVjdGx5JywgKCkgPT4ge1xuICBjb25zdCB0cmVlID0gcmVuZGVyZXIuY3JlYXRlKDxCdXR0b25BY3Rpb25Sb3VuZCAvPikudG9KU09OKCk7XG4gIGV4cGVjdCh0cmVlKS50b01hdGNoU25hcHNob3QoKTtcbn0pO1xuIl19