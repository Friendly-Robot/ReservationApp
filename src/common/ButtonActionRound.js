import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';

import theme from '../theme';

type buttonProps = {
  icon: string,
  onPress: Function,
};

export default ButtonActionRound = ({ onPress = () => {}, icon = '+' }: buttonProps) => (
  <View style={styles.container}>
    <TouchableOpacity
      activeOpacity={0.6}
      onPress={onPress}
      style={styles.button}  
    >
      <Text style={styles.icon}>{ icon }</Text>
    </TouchableOpacity>
  </View>
);

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    backgroundColor: theme.COLOR_PRIMARY,
    borderRadius: theme.BORDER_RADIUS_ROUND,
    ...theme.elevation,
    height: 50,
    justifyContent: 'center',
    width: 50,

  },
  container: {
    backgroundColor: 'transparent',
    bottom: 25,
    position: 'absolute',
    right: 15,
    zIndex: 10,
  },
  icon: {
    color: theme.COLOR_WHITE,
    fontSize: 35,
  },
});