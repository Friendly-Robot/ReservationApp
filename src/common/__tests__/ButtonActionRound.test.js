/**
 * @format
 */

import 'react-native';
import React from 'react';
import ButtonActionRound from '../ButtonActionRound';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<ButtonActionRound />', () => {
  it('should be defined', () => {
    expect(ButtonActionRound).toBeDefined();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ButtonActionRound />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
