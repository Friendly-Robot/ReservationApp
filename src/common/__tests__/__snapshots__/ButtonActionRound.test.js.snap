// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonActionRound /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "bottom": 25,
      "position": "absolute",
      "right": 15,
      "zIndex": 10,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(30, 147, 242)",
        "borderRadius": 1000,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "shadowColor": undefined,
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 2,
        "width": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(255, 255, 255)",
          "fontSize": 35,
        }
      }
    >
      +
    </Text>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "bottom": 25,
      "position": "absolute",
      "right": 15,
      "zIndex": 10,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(30, 147, 242)",
        "borderRadius": 1000,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "shadowColor": undefined,
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 2,
        "width": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(255, 255, 255)",
          "fontSize": 35,
        }
      }
    >
      +
    </Text>
  </View>
</View>
`;
